### PR TITLE
[10.x] Add `percentage` method to Collections

### PR DESCRIPTION
**Pull request by WendellAdriel**
## Changes

In this PR I add a new method called `percentage` to the Collections that allows us to calculate the percentage in a collection based on a truth test.

## Why

In many apps, we need to calculate percentage values of collections for something, especially when dealing with reports, analytics, metrics etc. Although that's something easy to calculate, doing it manually every time depending on the type of application we build is tiresome.

This method is just a life improvement for the DX, so we don't need to calculate that every time we need.

### CODE REVIEW

**Comment by WendellAdriel** [comment]
I saw that there's one failing check, but I checked the details and it doesn't seem to be related to the changes of the PR. There are also 2 other checks that were cancelled but IDK why. Should I do anything about these?

File: src/Illuminate/Collections/Collection.php [hunk 1]
```diff
@@ -1758,4 +1758,23 @@ public function offsetUnset($key): void
...
+     * Calculates the percentage of items that pass a given truth test.
+     *
+     * @param  (callable(TValue, TKey): bool)  $callback
+     * @param  int  $precision
+     * @return float
+     */
+    public function percentage(callable $callback, int $precision = 2): float
+    {
+        if ($this->isEmpty()) {
+            return 0;
```
Comment (cosmastech) [inline src/Illuminate/Collections/Collection.php:1772]: In projects I've worked on, I always have argued in favor of **not** returning 0 from functions like this because the computation never ran. I would normally return null to indicate that this is an exceptional case. That way I have a distinction between 0/10 meet the callback, rather than 0/0 meet the callback.

I realize it's easy enough to call `$collection->isEmpty()` before running this, and ultimately you have to check afterwards if the value is null or a proper float. 🤷

Comment (WendellAdriel) [reply src/Illuminate/Collections/Collection.php:1772]: Oh, that's an interesting point, because it could have some cases where the percentage itself could be 0. I'll update it to return null, thank you so much! 💪

**Comment by WendellAdriel** [comment]
@cosmastech I updated the PR regarding your observation and also added some more test cases to cover when the percentage should be `0` or `null`.

**Comment by WendellAdriel** [comment]
Thank you so much for merging this @taylorotwell!!!
I forgot about the Enumerable Trait and duplicated the code 😅 
Thanks for fixing that!!! 🔥
